src/fips/indicators.rs

```diff
@@ -841,6 +850,51 @@ const FIPS_CHECKS: FipsChecks = FipsChecks {
                 | CKF_UNWRAP
                 | CKF_DERIVE,
         },
+        FipsMechanism {
+            mechanism: CKM_TLS12_MASTER_KEY_DERIVE,
+            operations: CKF_DERIVE,
+            restrictions: [
+                restrict!(CKK_GENERIC_SECRET, step!(48 * 8)),
+                restrict!(),
+            ],
+            genflags: CKF_SIGN
+                | CKF_VERIFY
+                | CKF_ENCRYPT
+                | CKF_DECRYPT
+                | CKF_WRAP
+                | CKF_UNWRAP
+                | CKF_DERIVE,
+        },
+        FipsMechanism {
+            mechanism: CKM_TLS12_KEY_AND_MAC_DERIVE,
+            operations: CKF_DERIVE,
+            restrictions: [
+                restrict!(CKK_GENERIC_SECRET, range!(112, 255 * 64 * 8)),
+                restrict!(),
+            ],
+            genflags: CKF_SIGN
+                | CKF_VERIFY
+                | CKF_ENCRYPT
+                | CKF_DECRYPT
+                | CKF_WRAP
+                | CKF_UNWRAP
+                | CKF_DERIVE,
```

Comment:
Same Q as above

Reply:
The pkcs11 specs describe this that resulting keys have same usage as above, sign/verify/encrypt/decrypt/derive. The problem is that it generates two different keys, each with different usage, and this is something we can not catch here, I think:

>  The two MACing keys ("client_write_MAC_secret" and "server_write_MAC_secret") (if present) are always given a type of CKK_GENERIC_SECRET. They are flagged as valid for signing and verification.

> The other two keys ("client_write_key" and "server_write_key") are typed according to information found in the template sent along with this mechanism during a C_DeriveKey function call.  By default, they are flagged as valid for encryption, decryption, and derivation operations.

The wrap/unwrap is nonsense. Will remove.